modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -6420,6 +6396,36 @@ private void processCustomMessage(TcpDiscoveryCustomEventMessage msg, boolean wa
             }
         }
 
+        /**
+         * If new node is in the progress of being added we must store and resend undelivered messages.
+         *
+         * @param msg Processed message.
+         * @return {@code true} If message was appended to pending queue.
+         */
+        private boolean posponeUndeliveredMessages(final TcpDiscoveryCustomEventMessage msg) {
+            boolean joiningEmpty;
+
+            synchronized (mux) {
+                joiningEmpty = joiningNodes.isEmpty();
+
+                if (log.isDebugEnabled())
```

Comment:
If I understand the code correctly here we will always print to debug logs this message as no check for actual value of `joiningEmpty` variable is made.

Reply:
thanks! i fixed this issue.